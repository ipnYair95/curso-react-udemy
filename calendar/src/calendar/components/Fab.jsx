import { addHours } from "date-fns";
import { useCalendarStore, useUiStore } from "../../hooks"


export const Fab = () => {

  const { openDateModal } = useUiStore();

  const { setActiveEvent } = useCalendarStore();

  const handleClickNew = () => {

    setActiveEvent({
      _id: null,
      title: '',
      notes: '',
      start: new Date(),
      end: addHours(new Date(), 2),
      bgColor: '#fafafa',
      user: {
        _id: '123',
        name: 'Fernando'
      }
    });

    openDateModal();
    
  }

  return (
    <button className="btn btn-primary fab" onClick={handleClickNew} >
      <i className="fas fa-plus"></i>
    </button>
  )

}
