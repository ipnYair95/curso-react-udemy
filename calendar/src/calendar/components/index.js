export * from './Navbar';
export * from './CalendarEvent';
export * from './CalendarModal';
export * from './Fab';
export * from './FabDelete';