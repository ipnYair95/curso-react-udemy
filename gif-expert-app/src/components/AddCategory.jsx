import PropTypes from 'prop-types';
import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {

	const [inputValue, setInputValue] = useState("");

	const onInputChanged = ({ target }) => {
		setInputValue(target.value);
	};

	const onSubmit = (event) => {
		event.preventDefault();

		if (inputValue.trim().length === 0) {
			return;
		}
		
		setInputValue("");
		onNewCategory(inputValue.trim());
	};

	return (
		<form onSubmit={(event) => onSubmit(event)} aria-label="form" >
			<input
				type="text"
				placeholder="Buscar gifs"
				value={inputValue}
				onChange={onInputChanged}
			/>
		</form>
	);

};


AddCategory.propTypes = {
	onNewCategory: PropTypes.func.isRequired
};