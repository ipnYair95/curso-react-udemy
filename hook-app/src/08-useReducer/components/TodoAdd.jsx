import { useState } from "react";
import { useForm } from "../../hooks/useForm";

export const TodoAdd = ({ onNewTodo }) => {

  const { description, onInputChange, onResetForm } = useForm({
    description: ''
  });

  const onSubmit = (event) => {

    event.preventDefault();

    if (description.trim() === '') {
      return;
    }

    const newTodo = {
      id: new Date().getTime(),
      description: description.trim(),
      done: false
    };

    onNewTodo(newTodo);
    onResetForm();

  }

  return (
    <form onSubmit={onSubmit} >

      <input
        type="text"
        placeholder="¿Qué hay que hacer?"
        className="form-control"
        name="description"
        value={description}
        onChange={onInputChange}
      />

      <button type="submit" className="btn btn-outline-primary mt-1" >
        Agregar
      </button>

    </form>
  )
}
