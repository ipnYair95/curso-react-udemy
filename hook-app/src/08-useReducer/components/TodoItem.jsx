export const TodoItem = ({ task, onHandleDeleteTodo, onToogleTodo }) => {
  return (
    <>
      <li key={task.id} className="list-group-item d-flex justify-content-between" >
        <span
          className={`align-self-center ${(task.done) ? 'text-decoration-line-through' : ''}  `}
          onClick={() => onToogleTodo(task.id)}
          aria-label="span"
        >
          {task.description}
        </span>
        <button className=" btn btn-danger " onClick={() => onHandleDeleteTodo(task.id)} > Borrar </button>
      </li>
    </>
  )
}
