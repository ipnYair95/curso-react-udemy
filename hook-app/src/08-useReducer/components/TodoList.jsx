import { TodoItem } from "./"


export const TodoList = ({ tasks = [], onHandleDeleteTodo, onToogleTodo }) => {
  return (
    <>

      <ul className="list-group" >
        {tasks.map(task => (
          <TodoItem key={task.id}
            task={task}
            onHandleDeleteTodo={onHandleDeleteTodo}
            onToogleTodo={onToogleTodo}
          />))}
      </ul>

    </>
  )
}
