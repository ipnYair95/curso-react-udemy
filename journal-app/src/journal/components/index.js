export * from './Navbar.jsx';
export * from './SideBar.jsx';
export * from './ImageGallery.jsx';
export * from './SiderBarItem.jsx';