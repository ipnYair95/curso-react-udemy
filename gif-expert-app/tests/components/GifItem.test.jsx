import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GifItem', () => {

    const props = {
        title: 'title',
        url: 'https://one-punch.com/saitama.jpg'
    };

    test('should snapshot', () => {

        const { container } = render(<GifItem  {...props} />);

        expect(container).toMatchSnapshot();

    });

    test('debe mostrar la imagen con el URl y el ALT indicado', () => {

        render(<GifItem title={props.title} url={props.url} />);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(props.url);
        expect(alt).toBe(props.title);

    });

    test('debe mostrar el titulo en el componente', () => {

        render(<GifItem title={props.title} url={props.url} />);       

        expect( screen.getByText( props.title ).innerHTML ).toContain(props.title); 

    });



});