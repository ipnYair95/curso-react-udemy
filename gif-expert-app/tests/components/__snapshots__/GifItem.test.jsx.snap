// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
       
      title
       
    </p>
  </div>
</div>
`;
